### PR TITLE
chore: bump snappy to `27f34a580be4a3becf5f8c0cba13433f53c21337`

### PR DESCRIPTION
**Pull request by thalesmg**
To fix build errors in MacOS

```
[ 11%] Building CXX object CMakeFiles/snappy.dir/snappy-stubs-internal.cc.o
[ 13%] Building CXX object CMakeFiles/snappy.dir/snappy.cc.o
/Users/runner/work/erlang-rocksdb/erlang-rocksdb/deps/snappy/snappy.cc:1293:37: error: comparison of integers of different signs: 'unsigned long' and 'ptrdiff_t' (aka 'long') [-Werror,-Wsign-compare]
             (op + deferred_length) < op_limit_min_slop);
              ~~~~~~~~~~~~~~~~~~~~  ^ ~~~~~~~~~~~~~~~~~
/Users/runner/work/erlang-rocksdb/erlang-rocksdb/deps/snappy/snappy.cc:1293:37: error: comparison of integers of different signs: 'unsigned long' and 'ptrdiff_t' (aka 'long') [-Werror,-Wsign-compare]
             (op + deferred_length) < op_limit_min_slop);
              ~~~~~~~~~~~~~~~~~~~~  ^ ~~~~~~~~~~~~~~~~~
/Users/runner/work/erlang-rocksdb/erlang-rocksdb/deps/snappy/snappy.cc:1405:15: note: in instantiation of function template specialization 'snappy::DecompressBranchless<char *>' requested here
              DecompressBranchless(reinterpret_cast<const uint8_t*>(ip),
              ^
/Users/runner/work/erlang-rocksdb/erlang-rocksdb/deps/snappy/snappy.cc:1600:17: note: in instantiation of function template specialization 'snappy::SnappyDecompressor::DecompressAllTags<snappy::SnappyIOVecWriter>' requested here
  decompressor->DecompressAllTags(writer);
                ^
```

### CODE REVIEW

File: deps/snappy/.github/workflows/build.yml
Comment (id) [inline deps/snappy/.github/workflows/build.yml:1]: do we need this one?

File: deps/snappy/BUILD.bazel
Comment (id) [inline deps/snappy/BUILD.bazel:1]: do we use this?

File: deps/snappy/.github/workflows/build.yml
Comment (thalesmg) [reply deps/snappy/.github/workflows/build.yml:1]: I just overwrote the contents of the directory with the whole commit contents.  I could remove that, sure.

File: deps/snappy/BUILD.bazel
Comment (thalesmg) [reply deps/snappy/BUILD.bazel:1]: I just overwrote the contents of the directory with the whole commit contents. I could remove that, sure.

**Review by id — APPROVED** [review]

**Review by id — APPROVED** [review]

**Pull request merged**